decoder.go

```diff
@@ -0,0 +1,133 @@
+package drum
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+	"log"
+	"os"
+	"strings"
+)
+
+// DecodeFile decodes the drum machine file found at the provided path
+// and returns a pointer to a parsed pattern which is the entry point to the
+// rest of the data.
+func DecodeFile(path string) (*Pattern, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	p := &Pattern{}
+	if err := NewDecoder(file).Decode(p); err != nil {
+		log.Fatal(err)
+		return nil, err
+	}
+
+	return p, nil
+}
+
+// Decoder reads and decodes splice patterns from an input stream
+type Decoder struct {
+	io.Reader
+}
+
+// NewDecoder returns a new decoder that reads from r
+func NewDecoder(r io.Reader) *Decoder {
+	return &Decoder{r}
+}
+
+// Decode reads the input stream and decodes it's values into a Pattern
+func (d *Decoder) Decode(p *Pattern) error {
+	n, err := d.spliceHeaderInfo()
+	if err != nil {
+		if err != io.EOF && err != io.ErrUnexpectedEOF {
+			return errors.New("unable to decode .splice file header: " + err.Error())
```

Comment:
I think `fmt.Sprintf` is more common for creating strings. Also, would it make sense to propagate the error here? I don't know if wrapping the error adds anything, especially since the error's not a constant here.